### PR TITLE
Replace deprecated punycode with userland package

### PR DESCRIPTION
**Pull request by philoserf**
closes: #967 and #945
obsoletes: #513

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -11,7 +11,7 @@ var ParserBlock  = require('./parser_block');
 var ParserInline = require('./parser_inline');
 var LinkifyIt    = require('linkify-it');
 var mdurl        = require('mdurl');
-var punycode     = require('punycode');
+var punycode     = require('punycode/');
```
Comment (philoserf) [inline lib/index.js:14]: My code change

File: package.json [hunk 2]
```diff
@@ -41,6 +41,7 @@
     "entities": "~3.0.1",
     "linkify-it": "^4.0.1",
     "mdurl": "^1.0.1",
+    "punycode": "^2.3.1",
```
Comment (philoserf) [inline package.json:44]: The result of `npm add`.

File: dist/markdown-it.js [hunk 3]
```diff
@@ -7455,32 +7455,31 @@
 	 * Override to modify basic RegExp-s.
 	 **/  LinkifyIt.prototype.onCompile = function onCompile() {};
   var linkifyIt = LinkifyIt;
-  /*! https://mths.be/punycode v1.4.1 by @mathias */
-  /** Highest positive signed 32-bit float value */  var maxInt = 2147483647;
+  /** Highest positive signed 32-bit float value */  const maxInt = 2147483647;
```
Comment (philoserf) [inline dist/markdown-it.js:7458]: The result of `npm run-script browserify`.

File: dist/markdown-it.min.js
Comment (philoserf) [inline dist/markdown-it.min.js:1]: The result of `npm run-script browserify`.

**Review by philoserf — COMMENTED** [review]
Noted the source of changes

**Comment by bcomnes** [comment]
Might be easier to land if you only stage the punycode change and omit the `var` -> `let`/`const` changes, totally up to @rlidwka and @puzrin though!

**Comment by Martiix** [comment]
I need this fix, it's blocking me right now as I can't easily lock the node version in my project, let me know if I can do anything to help.

**Comment by philoserf** [comment]
The generated code change is part of the punycode change. I don't know who uses it but it is generated code in this project.

**Comment by puzrin** [comment]
Done in master after rewrite to ESM

**Pull request closed without merging**